resources/codegen_inputs/avar.rs

```diff
@@ -0,0 +1,34 @@
+#![parse_module(read_fonts::tables::avar)]
+
+/// The [avar (Axis Variations)](https://docs.microsoft.com/en-us/typography/opentype/spec/avar) table
+table Avar {
+    /// Major version number of the axis variations table — set to 1.
+    /// Minor version number of the axis variations table — set to 0.
+    version: MajorMinor,
+    /// Permanently reserved; set to zero.
+    #[skip_getter]
+    _reserved: u16,
+    /// The number of variation axes for this font. This must be the same number as axisCount in the 'fvar' table.
+    axis_count: u16,
+    /// The segment maps array — one segment map for each axis, in the order of axes specified in the 'fvar' table.
+    #[count(..)]
+    #[traverse_with(traverse_segment_maps)]
+    axis_segment_maps: VarLenArray<SegmentMaps<'a>>,
+}
+
+/// [SegmentMaps](https://learn.microsoft.com/en-us/typography/opentype/spec/avar#table-formats) record
+table SegmentMaps {
```

Comment:
this is supposed to be a record, not a table (and thank the lord, this was going to break a bunch of things)

Reply:
yeah, if I make it a record, it crashes codegen with 'array tokens never cooked'. Table just worked so I ran with it

Reply:
I get the complication in that tables imply some form of offset resolution, but it seems like anything inside a VarLenArray will require special casing regardless?

Reply:
it's just a really funny case. The basic issue is that we don't bother implementing `FontRead` for records, because in general they always exist in arrays, and we transmute, but `VarLenArray` uses `FontRead` under the covers.

The reason I ran into this is because it should be easy for this to work with traversal, *assuming* that the inner types are records. It doesn't work if they're tables, at least right now, because we always expect tables to be behind offsets.

As I write this, I think the best course of action right now would just be a manual impl of `FontRead` for this particular case; then at least we get traversal working.

The reason traversal is particularly tricky in `post` is because we don't really have any generalized concept of a 'string', so it isn't obvious how to even represent that as a FieldType.

I'm going to add a commit here if that's okay that hopefully gets this working, with traversal?

Reply:
By all means. Hack away!